src/HtmlGenerator/Utilities/Paths.cs

```diff
@@ -166,7 +166,9 @@ public static string GetRelativeFilePathInProject(Document document)
             string result = Path.Combine(document.Folders
                 .Select(SanitizeFolder)
                 .ToArray());
-            result = Path.Combine(result, document.Name);
+
+            result = Path.Combine(result, Path.GetFileName(document.FilePath));

```

Comment:
document.FilePath can be null in certain cases (Metadata As Source). Can you do var fileName = document.FilePath != null ? Path.GetFileName(document.FilePath) : document.Name or something?